src/main/kotlin/uk/gov/dwp/dataworks/Model.kt

```diff
@@ -43,3 +47,67 @@ data class UserTask(val correlationId: String,
         }
     }
 }
+
+@JsonDeserialize(builder = StatementObject.StatementObjectBuilder::class)
+data class StatementObject(var Sid: String, var Effect: String, var Action: List<String>, var Resource: List<String>) {
```

Comment:
I think you can use the Jackson Kotlin plugin to construct objects based on simple data classes with annotations. This should mean we can avoid having builders inside all of our data classes that need serialisation.
See [here](https://www.baeldung.com/jackson-kotlin#rename)

Reply:
Feel a bit silly having built out these objects with builders etc. and you can just use annotations... my bad!